File: cloudflare-worker/src/push-provider-senders/pushplus.ts
import type { NotificationMessage, ProviderConfig, PushResult } from "../types";
import { postJson } from "../push-http";

export async function sendPushPlus(
  provider: ProviderConfig,
  message: NotificationMessage,
  timeoutSec: number
): Promise<PushResult> {
  const payload: Record<string, unknown> = {
    token: provider.config.token,
    title: message.title,
    content: message.markdown,
    template: "markdown",
  };
  for (const key of ["topic", "channel"]) {
    const v = (provider.config[key] || "").trim();
    if (v) payload[key] = v;
  }
  return postJson(provider, "https://www.pushplus.plus/send", payload, timeoutSec, {
    successCodes: new Set([200, "200", 0, "0"]),
  });
}

File: src/roco_serverchan_notifier/push_provider_senders/pushplus.py
from __future__ import annotations

from ..push_http import HttpSession, JsonPostRequest, post_json
from ..push_models import NotificationMessage, ProviderConfig, PushResult


def send_pushplus(
    provider: ProviderConfig, message: NotificationMessage, session: HttpSession, timeout: int
) -> PushResult:
    payload = {
        "token": provider.config["token"],
        "title": message.title,
        "content": message.markdown,
        "template": "markdown",
    }
    for key in ("topic", "channel"):
        value = str(provider.config.get(key, "")).strip()
        if value:
            payload[key] = value
    return post_json(
        JsonPostRequest(
            provider,
            session,
            "https://www.pushplus.plus/send",
            payload,
            timeout,
            success_codes={200, "200", 0, "0"},
        )
    )
